modules/st2-auto-form/fields/array.js

```diff
@@ -78,13 +78,26 @@ export default class ArrayField extends BaseTextField {
       return v;
     }
 
+    /* YAQL parameter that came input as single yaql */
+    if (isYaql(v) && !v.includes(',')) {
```

Comment:
What happens if there is a comma inside the YAQL expression?

Reply:
It will fall down to the split below. So it will go direct to the split below on the ,.

So solving "," in yaql is another issue, as if you have commas in your strings, or yaqls - it's going to go wrong at that split function.  

So I wasn't going to try and solve commas at this stage, as that will require a whole host more work. I think there's going to be edge cases that it doesn't split correctly, but trying to cope with simple YAQL as per the default ST2 pack installs etc, it should.